pkg/k8s/k8s_cluster.go

```diff
@@ -449,3 +489,33 @@ func (k *K8sCluster) ProxyGet(scheme, namespace, name, port, path string, params
 	}
 	return ret.Stream(k.ctx)
 }
+
+func (k *K8sCluster) GetObjectStatus(ref k8s.ObjectRef) (string, error) {
+	o, _, err := k.GetSingleObject(ref)
+	if o == nil {
+		return "", err
+	}
+	conditions, ok, err := o.GetNestedField("status", "conditions")
+	if !ok {
+		return "", err
+	}
+	last := conditions.([]interface{})[0].(map[string]interface{})["status"]
+
+	return fmt.Sprintf("%s", last), err
+}
+
+func (k *K8sCluster) WaitForReady(ref k8s.ObjectRef) (bool, error) {
```

Comment:
This (and GetObjectStatus) is also flux specific as it won't work for most other objects.